apps/marsha/templates/services/nginx/configs/marsha.conf.j2

```diff
@@ -61,6 +61,22 @@ server {
     try_files $uri @proxy_to_marsha_app;
   }
 
+  location /ws/ {
+    proxy_pass http://marsha-backend;
+
+    proxy_http_version 1.1;
+    proxy_set_header Connection "Upgrade";
+    proxy_set_header Upgrade $http_upgrade;
+
+    proxy_buffering off;
+    proxy_set_header Host $host;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+
+    proxy_read_timeout {{ marsha_nginx_websocket_proxy_read_timeout }};
+    proxy_send_timeout {{ marsha_nginx_websocket_proxy_send_timeout }};
+  }
+
```

Comment:
IMO you don't need a new location if you allow to upgrade connections for the `/` location.

Reply:
This new location is needed to isolate websocket connection to others. For a classic http connection we don't want to allow a so big timeout. Only for websocket. So we must add this new location dedicated to websocket.

Reply:
One other argument to keep this location block is `http/2`. The UvicornWorker is able to handle http2 and I see that browsers compatible are using this protocol